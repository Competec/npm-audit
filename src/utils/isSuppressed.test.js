const {JestMockT} = require('jest');
const isSuppressed = require('./isSuppressed');
const logger = require('./logger');

jest.mock('./logger');
const mockedLogger = jest.mocked(logger);

test('should return true', () => {
    const suppressionList =
           [
               {
                   githubAadvisoryId: 'GHSA-hjp8-2cm3-cc45',
                   suppress: {
                       until: '2022-12-31Z',
                       reason: 'Third party',
                   },
               },
               {
                   githubAdvisoryId: 'GHSA-hhq3-ff78-jv3g',
                   suppress: {
                       until: '2023-03-31Z',
                       reason: 'eslint-loader, resolve-url-loader, react-dev-utils',
                   },
               },
           ];

    const auditAdvisory = {
        advisory: {
            findings: 'test',
            github_advisory_id: 'GHSA-hjp8-2cm3-cc45',
        },
    };
    expect(isSuppressed({suppressionList, auditAdvisory})).toBeTruthy();
});

test('should return false', () => {
    const suppressionList =
        [
            {
                githubAadvisoryId: 'GHSA-hjp8-2cm3-cc45',
                suppress: {
                    until: '2022-12-31Z',
                    reason: 'Third party',
                },
            },
        ];

    const auditAdvisory = {
        advisory: {
            findings: 'test',
            github_advisory_id: 'id',
        },
    };

    expect(isSuppressed({suppressionList, auditAdvisory})).toBeFalsy();
});

test('should return a logger with info: An advisory has been found...', () => {
    const suppressionList =
        [
            {
                githubAadvisoryId: 'GHSA-hjp8-2cm3-cc45',
            },
        ];

    const auditAdvisory = {
        advisory: {
            findings: 'test',
            github_advisory_id: 'GHSA-hjp8-2cm3-cc45',
        },
    };

    isSuppressed({suppressionList, auditAdvisory});
    expect(mockedLogger.warn).toMatchSnapshot();
});

test('should return a logger with info: Suppression has expired...', () => {
    const suppressionList =
        [
            {
                githubAadvisoryId: 'GHSA-hjp8-2cm3-cc45',
                suppress: {
                    until: '2021-12-31Z',
                    reason: 'Third party',
                },
            },
        ];

    const auditAdvisory = {
        advisory: {
            findings: 'test',
            github_advisory_id: 'GHSA-hjp8-2cm3-cc45',
        },
    };

    isSuppressed({suppressionList, auditAdvisory});
    expect(mockedLogger.warn).toMatchSnapshot();
});
