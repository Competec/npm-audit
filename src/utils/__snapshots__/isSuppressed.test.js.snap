// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return a logger with info: An advisory has been found... 1`] = `
[
  [
    "An advisory has been found. You can suppress this in .yarn-audit-competec/suppressions.js",
    {
      "findings": "test",
      "githubAdvisoryId": "id",
    },
  ],
  [
    "An advisory has been found. You can suppress this in .yarn-audit-competec/suppressions.js",
    {
      "findings": "test",
      "githubAdvisoryId": "GHSA-hjp8-2cm3-cc45",
    },
  ],
]
`;

exports[`should return a logger with info: Suppression has expired... 1`] = `
[
  [
    "An advisory has been found. You can suppress this in .yarn-audit-competec/suppressions.js",
    {
      "findings": "test",
      "githubAdvisoryId": "id",
    },
  ],
  [
    "An advisory has been found. You can suppress this in .yarn-audit-competec/suppressions.js",
    {
      "findings": "test",
      "githubAdvisoryId": "GHSA-hjp8-2cm3-cc45",
    },
  ],
  [
    "Suppression has expired!",
    {
      "findings": "test",
      "githubAdvisoryId": "GHSA-hjp8-2cm3-cc45",
    },
  ],
]
`;
